clusterloader2/pkg/measurement/util/wait_for_pvcs.go

```diff
@@ -0,0 +1,92 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+// WaitForPVCOptions is an options used by WaitForPVCs methods.
+type WaitForPVCOptions struct {
+	Selector            *ObjectSelector
+	DesiredPVCCount     int
+	EnableLogging       bool
+	CallerName          string
+	WaitForPVCsInterval time.Duration
+}
+
+// WaitForPVCs waits till disire nuber of PVCs is running.
+// PVCs are be specified by namespace, field and/or label selectors.
+// If stopCh is closed before all PVCs are running, the error will be returned.
+func WaitForPVCs(clientSet clientset.Interface, stopCh <-chan struct{}, options *WaitForPVCOptions) error {
+	ps, err := NewPVCStore(clientSet, options.Selector)
+	if err != nil {
+		return fmt.Errorf("PVC store creation error: %v", err)
+	}
+	defer ps.Stop()
+
+	var pvcsStatus PVCStartupStatus
+	scaling := uninitialized
+	var oldPVCs []*corev1.PersistentVolumeClaim
+	for {
+		select {
+		case <-stopCh:
+			return fmt.Errorf("timeout while waiting for %d PVCs to be running in namespace '%v' with labels '%v' and fields '%v' - only %d found bound",
+				options.DesiredPVCCount, options.Selector.Namespace, options.Selector.LabelSelector, options.Selector.FieldSelector, pvcsStatus.Bound)
+		case <-time.After(options.WaitForPVCsInterval):
+			pvcs := ps.List()
+			pvcsStatus = ComputePVCStartupStatus(pvcs, options.DesiredPVCCount)
+			if scaling != uninitialized {
+				diff := DiffPVCs(oldPVCs, pvcs)
+				deletedPVCs := diff.DeletedPVCs()
+				if scaling != down && len(deletedPVCs) > 0 {
+					klog.Errorf("%s: %s: %d PVCs disappeared: %v", options.CallerName, options.Selector.String(), len(deletedPVCs), strings.Join(deletedPVCs, ", "))
+					klog.Infof("%s: %v", options.CallerName, diff.String(sets.NewString()))
+				}
+				addedPVCs := diff.AddedPVCs()
+				if scaling != up && len(addedPVCs) > 0 {
+					klog.Errorf("%s: %s: %d PVCs appeared: %v", options.CallerName, options.Selector.String(), len(deletedPVCs), strings.Join(deletedPVCs, ", "))
+					klog.Infof("%s: %v", options.CallerName, diff.String(sets.NewString()))
+				}
+			} else {
+				switch {
+				case len(pvcs) == options.DesiredPVCCount:
+					scaling = none
+				case len(pvcs) < options.DesiredPVCCount:
+					scaling = up
+				case len(pvcs) > options.DesiredPVCCount:
+					scaling = down
+				}
+			}
+			if options.EnableLogging {
```

Comment:
what about all the other times we log things, why is only this one gated by `EnableLogging`? There should probaly be a wrapper function for this.

Reply:
The other are error logs, and this one particular log info. Not only that, it is an info that is once per iteration of the for loop so it can become quite annoying if youre waiting for a long time so it makes sense that this one should have an option to be disabled.